src/movies.js

```diff
@@ -1,22 +1,99 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
+
+const movies = require("./data")
+
+function getAllDirectors() {
+  const directors = movies.map(function (film) {
+    return film.director
+    })
+  return directors
+  }
+  console.log(getAllDirectors(movies).length) // => 250
+
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectorsUniq() {
+  const directors = movies.map(function (film) {
+    return film.director
+    }).reduce(function(a,b){
+      if (a.indexOf(b) < 0 ) a.push(b)
+      return a
+    },[])
+  return directors
+  }
+console.log(getAllDirectorsUniq(movies).length) // => 149
```

Comment:
It is easier if you do it with `filter` or `set`. Check the solution posted by Jan.